File: 알고리즘 문제풀이/기타/CombinationSum.py
    
def combinationSum(self, candidates: List[int], target: int) -> List[List[int]]:
    res=[]
    def backtrack(target,n_set,idx):
        if sum(n_set)==target:
            res.append([ele for ele in n_set])
            return
        elif sum(n_set)<target and idx<len(candidates):
            n_set.append(candidates[idx])
            backtrack(target,n_set,idx)
            n_set.pop()
            backtrack(target,n_set,idx+1)
    backtrack(target,[],0)
    return res

File: 알고리즘 문제풀이/LeetCode/combinationSum.js
/**
 * https://leetcode.com/problems/combination-sum/submissions/
 * @param {number[]} candidates
 * @param {number} target
 * @return {number[][]}
 */
const combinationSum = function (candidates, target) {
  const ans = [];

  const backtrack = (arr, total, idx) => {
    if (total > target) return;

    if (total === target) {
      ans.push(arr);
      return;
    }

    for (let i = idx; i < candidates.length; i++) {
      const num = candidates[i];
      backtrack([...arr, num], total + num, i);
    }
  };

  for (let i = 0; i < candidates.length; i++) {
    const num = candidates[i];
    backtrack([num], num, i);
  }
  return ans;
};
